acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/privacy/condition/ExpectValidPrivateContractDeployedReceipt.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.tests.acceptance.dsl.privacy.condition;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.Contract;
+
+public class ExpectValidPrivateContractDeployedReceipt implements PrivateContractCondition {
+
+  private final String contractAddress;
```

Comment:
should these both be of type Address?

Reply:
nope - I am of the opinion that at this level of the dsl they should correspond to the types returned by web3j